test/com/facebook/buck/jvm/java/JavaOrderingIntegrationTest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2018-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.jvm.java;
+
+import com.facebook.buck.jvm.java.testutil.AbiCompilationModeTest;
+import com.facebook.buck.testutil.ProcessResult;
+import com.facebook.buck.testutil.TemporaryPaths;
+import com.facebook.buck.testutil.integration.ProjectWorkspace;
+import com.facebook.buck.testutil.integration.TestDataHelper;
+import com.google.common.base.Joiner;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.zip.ZipFile;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+
+/**
+ * See https://github.com/facebook/buck/issues/1830 for background.
+ *
+ * Integration test to verify:
+ * 1. Java tests and java binaries access parameters in the same order.
+ * 2. Order of the resolution.
+ * Changing resolution order is breaking change and this test will catch it.
+ */
+public class JavaOrderingIntegrationTest extends AbiCompilationModeTest {
+
+  private static final String JAVA_RESOLUTION_ORDER = Joiner.on(",").join(
+      "1-2.txt:1",
+      "1-3.txt:1",
+      "1-4.txt:1",
+      "1-5.txt:1",
+      "2-3.txt:2",
+      "2-4.txt:2",
+      "2-5.txt:2",
+      "3-4.txt:3",
+      "3-5.txt:3",
+      "4-5.txt:4");
+
+  private static final String TEST_OUTPUT_MARKER = "===+===";
+
+  @Rule
+  public TemporaryPaths tmp = new TemporaryPaths();
+
+  @Test
+  public void testJavaBinaryOrdering() throws IOException {
```

Comment:
Please, add more information to the method name to indicate what exactly is testing in this method.